client/state/data-layer/wpcom-http/index.js

```diff
@@ -0,0 +1,60 @@
+/**
+ * External dependencies
+ */
+import { get, noop } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import wpcom from 'lib/wp';
+import { WPCOM_HTTP_REQUEST } from 'state/action-types';
+import { extendAction } from 'state/utils';
+
+/**
+ * Returns the appropriate fetcher in wpcom given the request method
+ *
+ * fetcherMap :: String -> (Params -> Query -> [Body] -> Promise)
+ *
+ * @param {string} method name of HTTP method for request
+ * @returns {Function} the fetcher
+ */
+const fetcherMap = method => get( {
```

Comment:
> WordPress.com API doesn't currently allow anything but POST and GET requests and wpcom.js actually just aliasing req.put and req.del as req.post

That's right. We added these aliases thinking on that the request object (`req.post`, `req.put`, etc ...) could act as an abstraction layer and in this way offering to the user all methods tied to the HTTP verbs. It could allow us to be consistent when we need to add a new connection with other REST APIs, for instance, with WP REST API (.org) which has all HTTP verbs.